spec/sidekiq/cancel_reservations_job_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe CancelReservationsJob, type: :job do
+  describe 'perform_now' do
+    let(:reservation) { create :reservation, seance_id: seance.id }
+    context 'when seance starts less than 30 minutes from now' do
+      let(:seance) { create :seance, start_time: 25.minute.from_now }
+      it 'cancels reservations' do
+        expect { described_class.perform_now }.to change { reservation.reload.status }.from('reserved').to('canceled')
+      end
+    end
+
+    context 'when seance starts 30 minutes from now' do
+      let(:seance) { create :seance, start_time: 30.minute.from_now }
+      it 'cancels reservations' do
+        expect { described_class.perform_now }.to change { reservation.reload.status }.from('reserved').to('canceled')
+      end
+    end
+
+    context 'when seance starts more than 30 minutes from now' do
+      let(:seance) { create :seance, start_time: 35.minute.from_now }
+      it 'does not cancel reservation' do
+        expect { described_class.perform_now }.not_to(change { reservation.status })
+      end
+    end
+  end
+
+  describe 'perform_later' do
+    it 'enqueues job' do
+      ActiveJob::Base.queue_adapter = :test
+      expect { described_class.perform_later }.to have_enqueued_job
+    end
+  end
```

Comment:
You don't really need to test that - you are basically testing the library here. `have_enqueued_job` is meant for places when you are actually calling a job - so not really applicable for cron jobs.

For example you might have: `ClearSeancesCacheJob` that will be scheduled in some controller, everytime somebody adds, changes, or removes a seance. Then when testing the controller you would use `have_enqueued_job` to check if this job is actually scheduled on that actions 😉

Reply:
We talked about this on Friday and it completely slipped my mind to delete it.